mbodied/agents/backends/httpx_backend.py

```diff
@@ -73,28 +73,74 @@ def __init__(
         self.headers = {"X-Api-Key": self.api_key, "Content-Type": "application/json"}
         self.serialized = serializer or self.SERIALIZER
         self.kwargs = kwargs
+        self.DEFAULT_MODEL = kwargs.get("model", self.DEFAULT_MODEL)
 
-    def predict(self, messages: List[Message], model: str | None = None, **kwargs) -> str:
+    @overload
+    def predict(self, messages: List[Message], model: str | None = None, **kwargs) -> str: ...
```

Comment:
This is kinda confusing. the message should not be a list right.

And why model = context?

if isinstance(context, str):
       model = context

and what does messages = [message] + messages do?

Reply:
Yep overloading in python is not great but type hints and interface simplification outweighs increased complexity